backend/src/main/java/heartbeat/service/report/LeadTimeForChangesCalculator.java

```diff
@@ -0,0 +1,85 @@
+package heartbeat.service.report;
+
+import heartbeat.client.dto.codebase.github.LeadTime;
+import heartbeat.client.dto.codebase.github.PipelineLeadTime;
+import heartbeat.controller.report.dto.response.AvgLeadTimeForChanges;
+import heartbeat.controller.report.dto.response.LeadTimeForChanges;
+import heartbeat.controller.report.dto.response.LeadTimeForChangesOfPipelines;
+import heartbeat.util.TimeUtil;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@Log4j2
+@RequiredArgsConstructor
+@Component
+public class LeadTimeForChangesCalculator {
+
+	public LeadTimeForChanges calculate(List<PipelineLeadTime> pipelineLeadTime) {
+		int pipelineCount = pipelineLeadTime.size();
+		List<LeadTimeForChangesOfPipelines> leadTimeForChangesOfPipelines = new ArrayList<>();
+		AvgLeadTimeForChanges avgLeadTimeForChanges = new AvgLeadTimeForChanges();
+
+		if (pipelineLeadTime.isEmpty()) {
+			return new LeadTimeForChanges(leadTimeForChangesOfPipelines, avgLeadTimeForChanges);
+		}
+
+		List<HashMap<String, Double>> avgDelayTimeMapList = pipelineLeadTime.stream().map(item -> {
+			int times = item.getLeadTimes().size();
```

Comment:
Maybe you should do `item.getLeadTimes().isEmpty()` before `item.getLeadTimes().size()`.

Reply:
will be fix in next commit